source/ota.c

```diff
@@ -1818,7 +1818,7 @@ static DocParseErr_t parseJSONbyModel( const char * pJson,
     const char * pFileParams = NULL;
     uint32_t fileParamsLength = 0;
 
-    LogDebug( ( "JSON received: %s", pJson ) );
+    LogDebug( ( "JSON received: %.*s", messageLength, pJson ) );
```

Comment:
This makes an even bigger assumption that LogDebug is implemented by printf.  We may as well just use printf.

Reply:
Thanks @n9wxu,
I updated the logic here to use a 256 bytes temp buffer to store the json because the buffer is provided by application which is not controlled by us.
To avoid big local buffer in the stack, I use static memory instead.

Please help to take a look again.

Thanks.